backend/main-service/src/main/java/kkakka/mainservice/product/domain/Product.java

```diff
@@ -1,4 +1,48 @@
 package kkakka.mainservice.product.domain;
 
+import kkakka.mainservice.category.domain.Category;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import org.hibernate.annotations.ColumnDefault;
+import javax.persistence.*;
+import java.util.Date;
+
+@Entity
+@Table(name = "product")
+@Getter
+@NoArgsConstructor
 public class Product {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    @Column(name = "PRODUCT_ID")
+    private Long id;
+
+    @ManyToOne(fetch = FetchType.LAZY)
+    private Category category;
+
+    @Column(nullable = false)
+    private String name;
+    private Integer price;
+    private Integer stock;
+
+    @Column(nullable = true)
+    private String image_url;
+    private String detail_image_url;
+    private String nutrition_info_url;
+    @ColumnDefault("0")
+    private Integer discount;
+
+    @Column(nullable = false, updatable = false, insertable = false)
+    @ColumnDefault(value = "CURRENT_TIMESTAMP")
+    private Date registered_at;
+
+    public Product(Category category, String name, int price, int stock, String image_url, String detail_image_url) {
+        this.category = category;
+        this.name = name;
+        this.price = price;
+        this.stock = stock;
+        this.image_url = image_url;
+        this.detail_image_url = detail_image_url;
+    }
```

Comment:
```suggestion
    public Product(Long id, Category category, String name, int price, int stock, String imageUrl, String detailImageUrl, String nutritionImageUrl) {
        this.id = id;
        this.category = category;
        this.name = name;
        this.price = price;
        this.stock = stock;
        this.imageUrl = imageUrl;
        this.detailImageUrl = detailImageUrl;
        this.nutritionInfoUrl = nutritionInfoUrl;
    }
    
    public Product(Category category, String name, int price, int stock, String image_url, String detail_image_url) {
        this(null, category, name, price, stock, imageUrl, detailImageUrl, null);
    }
```

이렇게 부생성자로 만들어보아요.
엘레강트 오브젝트 보면 부생성자를 많이 만들라는 얘기가 나오는걸로 기억하는디 저도 이 의견에 동의해요.